contacts/tests.py

```diff
@@ -1,3 +1,122 @@
+    seeker = create_seeker(user1, new_city, "2020-05-05", 100, 3000, 2, 2, "Hi")
+    seeker.save()
+    return seeker
+
+
+@pytest.fixture
+def seeker_2(new_city):
+    user2 = create_user("t2@m.com", "test2", "test", "1995-05-05", "testing")
+    seeker = create_seeker(user2, new_city, "2020-05-05", 300, 3000, 3, 3, "Hi")
+    seeker.save()
+    return seeker
+
+
+@pytest.fixture
+def apartment_1(new_city):
+    user3 = create_user("t3@m.com", "test3", "test", "1995-05-05", "testing")
+    apartment = create_apartment(user3, new_city, "Hatotahim 5", 2500, 2, 3, "2020-05-05")
+    apartment.save()
+    return apartment
+
+
+@pytest.fixture
+def connection_1(seeker_1, apartment_1):
+    con = Connection(seeker=seeker_1, apartment=apartment_1)
+    con.save()
+    return con
+
+
+@pytest.mark.django_db
+def test_add_connection(connection_1):
+    assert connection_1 is not None
+    assert connection_1.seeker.base_user.email == "t1@m.com"
+    assert connection_1.apartment.owner.email == "t3@m.com"
+
+
+@pytest.mark.django_db
+def test_find_connection(connection_1):
+    temp_connection = Connection.objects.get(seeker=connection_1.seeker, apartment=connection_1.apartment)
+    assert temp_connection == connection_1
+
+
+@pytest.mark.django_db
+def test_invalid_users_connection(seeker_1, seeker_2):
+    with pytest.raises(ValueError):
+        with transaction.atomic():
+            Connection(seeker=seeker_1, apartment=seeker_2).save()
+
+
+@pytest.mark.django_db
+def test_invalid_duplicate_connection(connection_1, seeker_1, apartment_1):
+    with pytest.raises(IntegrityError):
+        with transaction.atomic():
+            Connection(seeker=seeker_1, apartment=apartment_1).save()
+
+
+@pytest.mark.django_db
+def test_multipile_connection(connection_1, seeker_2, apartment_1):
+    Connection(seeker=seeker_2, apartment=apartment_1).save()
+    amount = Connection.objects.filter(apartment=apartment_1).count()
+    expected = 2
+    assert amount == expected
+
+
+@pytest.mark.django_db
+def create_user(email, f_name, l_name, b_date, passw):
+    user1 = User(
+        email=email,
+        first_name=f_name,
+        last_name=l_name,
+        birth_date=b_date,
+        password=passw
+    )
+    user1.save()
+    return user1
+
+
+def create_city():
+    city = City(cityName='Tel Aviv')
+    return city
+
+
+def create_seeker(user, city, s_date, min_r, max_r, roomates, rooms, about):
+    seeker = Seeker(
+        base_user=user,
+        city=city,
+        start_date=s_date,
+        min_rent=min_r,
+        max_rent=max_r,
+        num_of_roomates=roomates,
+        num_of_rooms=rooms,
+        about=about
+    )
+    return seeker
+
+
+def create_apartment(user, city, addr, rent, roomates, rooms, s_date):
+    apartmant = Apartment(
+        owner=user,
+        city=city,
+        address=addr,
+        rent=rent,
+        num_of_roomates=roomates,
+        num_of_rooms=rooms,
+        start_date=s_date
+    )
+    return apartmant
```

Comment:
Why did you add all those create functions? what is their purpose?

Reply:
There's a strong purpose,
I have to create multiple objects in order to create connections.
Those include creating an apartment, seeker, and each requires creating a user too.